### PR TITLE
Add execute-to-branch option

### LINKED ISSUE #2848: Add --execute-to-branch option
Edit: disregard description below, see https://github.com/kframework/kore/issues/2848#issuecomment-940610272 and the next comment.

------------
Given an execution, I'd like to make it easy to log (states + path conditions):

- The entry points (initial states)
- The exist points (final states, proven or unproven)
- The branch points (state before branch, each state after branch)
- Enough information to reconstruct the branching graph from these log entries.

Maybe the infrastructure for this already exists, but I'd like to make it easy to do something like `--log-branch-points`, because this functionality can be used to easily summarize the control-flow of an existing program, and to construct lemmas of each basic-block of execution (so we can step through them more quickly the next execution).

### PR DESCRIPTION
**Pull request by andreiburdusa**
Fixes #2848

### Estimate:
- March 28

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

**Comment by andreiburdusa** [comment]
Example:
```
$ KORE_EXEC_OPTS="--execute-to-branch" kprovex --haskell-backend-command /home/andrei/kore/.build/kore/bin/kore-exec sum-spec.k
kore-exec: [501283] Info (InfoProofDepth):
    proof incomplete: proof depth: 10
  <T>
    <k>
      if (notBool N <=Int 0) {
        {
          sum = sum + n;
          n = n + -1;
        }
        while (! n <= 0) {
          sum = sum + n;
          n = n + -1;
        }
      } else { } ~> _DotVar2 ~> .
    </k>
    <state>
      n |-> N:Int
      sum |-> S:Int
    </state>
  </T>
#And
  {
    true
  #Equals
    N >=Int 0
  }
```

**Review by ana-pantilie — APPROVED** [review]
Excellent! 👍  I like your idea of also logging the proof depth.

File: kore/app/share/GlobalMain.hs [hunk 1]
```diff
@@ -206,6 +208,12 @@ parseKoreProveOptions =
                         \in case the prover fails."
                 )
             )
+        <*> flag
+            Leaf
+            LeafOrBranching
+            ( long "execute-to-branch"
+                <> help "Execute untill the proof branches."
```
Comment (ehildenb) [inline kore/app/share/GlobalMain.hs:215]:
```suggestion
                <> help "Execute until the proof branches."
```

**Comment by ehildenb** [comment]
Just want to double-check that this is enabled for `kore-exec ...`, it's important that it's exposed without having to launch the kore-repl.

**Pull request merged**